torch/distributed/pipelining/PipelineStage.py

```diff
@@ -232,6 +233,16 @@ def map_recv_to_send(a):
         )
         return grad_send_info
 
+    def _prepare_backward_infra(self, num_microbatches: int):
+        # TODO: this is needed for backward_maybe_with_nosync
+        self.chunks = num_microbatches
```

Comment:
nit: why don't we set `self.chunks` earlier in `_prepare_forward_infra`?
`_prepare_forward_infra` is always called, while `_prepare_backward_infra` is conditionally called.

Reply:
self.chunks is only needed in backwards though for `_check_chunk_id` in `backward_maybe_with_nosync`

Reply:
True, although it could become a hidden surprise for future developers.
Is there a disadvantage in setting it early?